packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -620,6 +620,13 @@ class EditableText extends StatefulWidget {
   /// context, the English phrase will be on the right and the Hebrew phrase on
   /// its left.
   ///
+  /// When LTR text is entered into an RTL field, or RTL text is entered into an
+  /// LTR field, [LRM](https://en.wikipedia.org/wiki/Left-to-right_mark) or
+  /// [RLM](https://en.wikipedia.org/wiki/Right-to-left_mark) characters will be
+  /// inserted alongside whitespace characters, respectively. These characters
+  /// will affect the length of the string and may need to be parsed out when
+  /// doing things like string comparison with other text.
```

Comment:
Maybe add a slight justification for this so that it is a bit more understandable:

"This is to eliminate ambiguous directionality in whitespace and ensure proper caret placement"